arc-mlir/src/include/Rust/RustPrinterStream.h

```diff
@@ -139,8 +139,8 @@ class RustPrinterStream {
       if (t.isa<FunctionType>() || t.isa<types::RustEnumType>() ||
           t.isa<types::RustStructType>() || t.isa<types::RustTensorType>() ||
           t.isa<types::RustTupleType>())
-        return "v" + std::to_string(id) + ".clone()";
-      return "v" + std::to_string(id);
+        return "arcorn::val!(v" + std::to_string(id) + ")";
+      return "arcorn::val!(v" + std::to_string(id) + ")";
```

Comment:
With `acorn::val!` used for all types, special-casing the non-primitive types is no longer needed, so drop the `if`.

Reply:
Ok, it's an abstraction with "hopefully no regret" 😅